amberdata/src/endpoint/index.ts

```diff
@@ -0,0 +1,13 @@
+export * as balance from './balance'
+export * as price from './price'
+
+export const COIN_KEYS = ['btc', 'eth', 'bch', 'ltc', 'btsv', 'zec'] as const
+export type CoinType = typeof COIN_KEYS[number]
+export function isCoinType(key: string): key is CoinType {
+  return COIN_KEYS.includes(key as CoinType)
+}
+export const CHAIN_KEYS = ['mainnet', 'testnet'] as const
+export type ChainType = typeof CHAIN_KEYS[number]
+export function isChainType(key: string): key is ChainType {
+  return CHAIN_KEYS.includes(key as ChainType)
+}
```

Comment:
Let's try to reuse this and add it to `@chainlink/types`

Reply:
I don't think that will work in the way that I am using the type guards here because each adapter might support different coins or chains. We can't generate these dynamically with TypeScript as far as I know.

Reply:
Yeah, I know, but my thinking was that we should take over those specific types and make more general types for our "framework". We now have a specific `CoinType` for every adapter that supports some kind of coins, but we don't have a `CoinType` on the framework level, which I think would be useful. Maybe we need to type both the per adapter specific types and the general ones. 🤔 

Let's leave it like you defined it for now, and think about it some more about how to handle it in the future.